### PR TITLE
Switch to mobile API

### LINKED ISSUE #1928: ABC iView restricting number of shows in the feed
I've noticed that iView is providng only a small number of shows (1 or 2 in most cases) in the feed response. Any idea about what is causing this behaviour? Just a change in policy from ABC? I couldnt find documentation on their api.

Cheers,

### PR DESCRIPTION
**Pull request by glennguy**
Switch add-on to the android/ios iView API.  The general layout has been
for the most part retained although there are some differences in the
program categories.
I've borrowed the threading code from the plus7 add-on to speed up the
listing of episodes as we now have to fetch each episodes individual
page to get the metadata we want.

Tested on Win 10/Kodi Beta 1 and RPi3/LibreELEC 7.0.2

The only issue I've come up against I'm pretty sure is to do with how
Kodi handles and stores cookies. I've had to clear the cookes.dat file a
couple of times to get rid of HTTP 403 errors. If it ends up being a
recurring issue I can implement manual processing of the m3u8 and
cookies. See this thread on the Kodi forum for a similar example -
[http://forum.kodi.tv/showthread.php?tid=232439&pid=2057557]

fixed #1928

### CODE REVIEW

File: resources/lib/config.py [hunk 1]
```diff
@@ -42,20 +42,13 @@
...
-
-# Used for "SWF verification", a stream obfuscation technique
-swf_hash    = '96cc76f1d5385fb5cda6e2ce5c73323a399043d0bb6c687edd807e5c73c42b37'
-swf_size    = '2122'
-swf_url     = 'http://www.abc.net.au/iview/images/iview.jpg'
+base_url     = 'http://iview.abc.net.au'
+config_url   = 'http://iview.abc.net.au/api/navigation/mobile/2/?device=android-mobile&'
+auth_url     = '/auth/hls/sign?'
+index_url    = 'http://iview.abc.net.au/api/index?device=android-mobile&fields=seriesTitle,episodeCount,href'
+feed_url     = ' http://iview.abc.net.au/api/{0}?device=android-mobile&sort=az'
```
Comment (andybotting) [inline resources/lib/config.py]: whitespace

File: resources/lib/parse.py [hunk 2]
```diff
@@ -31,172 +31,133 @@
...
-            if s.title == title:
-               show = s
-               break
-
-        if show:
-            show.increment_num_episodes()
-        else:
+    href_list = []
+    show_index = json.loads(comm.fetch_url(config.index_url))
+    series_houseno_list = [(x['href'][-13:-6], x['episodeCount'])
```
Comment (andybotting) [inline resources/lib/parse.py]: can we split this in a smarter way?

File: resources/lib/parse.py [hunk 3]
```diff
@@ -31,172 +31,133 @@
...
+            subtitle = item['title']
+            p.title = title
+            p.episode_title = subtitle
+        else:
+            title = item['seriesTitle']
+            p.title = title           
+        
+        p.house_number  = item['episodeHouseNumber']
+        p.description   = item['description']
+        p.url           = item['playlist'][-1]['hls-high']
```
Comment (andybotting) [inline resources/lib/parse.py]: ['hls-high']: do we get a choice for other qualities?

File: resources/lib/parse.py [hunk 4]
```diff
@@ -31,172 +31,133 @@
...
             # Rating not given for all programs
             pass
 
         try:
-            duration = item.find('{http://search.yahoo.com/mrss/}content').attrib['duration']
+            duration = item['duration']
             p.duration = int(duration)
         except:
-            utils.log("Couldn't parse program duration: %s" % duration)
+            if 'duration' in locals():
```
Comment (andybotting) [inline resources/lib/parse.py]: how about something like `if 'duration' in item:` instead?

File: resources/lib/play.py [hunk 5]
```diff
@@ -38,11 +38,14 @@ def play(url):
...
-
+        auth = utils.get_auth(p)
+        akamai_auth = utils.get_akamai_auth(auth)
+        akamai_url = "{0}?hdnea={1}".format(p.get_url(), akamai_auth)
+        
         listitem=xbmcgui.ListItem(label=p.get_list_title(),
                                   iconImage=p.thumbnail,
                                   thumbnailImage=p.thumbnail,
-                                  path=p.get_url())
+                                  path = akamai_url)
```
Comment (andybotting) [inline resources/lib/play.py]: should remove spaces around `=` when passed as kwargs

**Comment by andybotting** [comment]
This looks nice @glennguy 

Have a look over some of my comments and let me know what you think

File: resources/lib/parse.py (hunk 3 above)
Comment (glennguy) [inline resources/lib/parse.py]: All of the stream options have multiple streams to chose from within. There is hls-base and hls-low but base is the same as high, and low just omits the 650kb/s stream, and when passed to Kodi as-is it seems from looking at logs that Kodi always likes to chose the highest quality anyway.

File: resources/lib/parse.py (hunk 4 above)
Comment (glennguy) [inline resources/lib/parse.py]: Wow I should have just done that to begin with!

File: resources/lib/parse.py (hunk 2 above)
Comment (glennguy) [inline resources/lib/parse.py]: Is it just an ugliness thing? Should I replace it with nested for loops?

Comment (andybotting) [inline resources/lib/parse.py]: I was referring more to the `[-13:-6]` substring selection. The list comprehension is fine

Comment (glennguy) [inline resources/lib/parse.py]: the href key contains eg. 'programs/arthur/ZX0564A020S00'

The episode house number is all 13 characters and the series house number
is the first 7 which seems to be in the format AB1234C - I don't really
know regex but would that be a better way to get it?

Comment (andybotting) [inline resources/lib/parse.py]: :+1: all good

**Comment by KODeKarnage** [comment]
Regex code:
`try:
    series_houseno_list  = re.findall(r"\/(\w{6})", s)[-1]
except IndexError:
   series_houseno_list  = FALLBACKVALUE `

**Comment by glennguy** [comment]
re: the cookies issues I have been experiencing - I set a proxy server in Kodi to debug from and it seems there's a bug where Kodi won't pass on headers and cookies over a proxy server. The manual stream handling circumvented that but it wasn't an ideal solution. So going forward we should be all set!

**Pull request merged**